torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -737,14 +738,15 @@ void SegmentedFusion::finalize() {
     }
 
     // Only look at ones that need to cast to fp16
-    if (force_fp16_tv_set_.count(edge_tv)) {
-      auto cast_tv_it = fp32_to_fp16_cast_map.find(edge->val->as<TensorView>());
+    if ((force_fp16_tv_set_.count(edge_tv)  > 0)) {
+      DataType dtype = edge_tv->getDataType().value();
```

Comment:
Hmm, this doesn't look right to me. If the tv is in `force_fp16_tv_set_`, very likely its data type will be `Float` but we want to cast to either `BFloat` or `Half`. 

The easiest way I could think of is to put a `force_half_precision_type_` in `SegmentedFusion` and set it in  `SegmentedFusion::annotateFP16IntermediateTensors`. Maybe simply just go through complete_fusion_->outputs() and set it to half if you see a half, bfloat if you see a bfloat. 

Please let me know if you'd need help with navigating the IR.